### PR TITLE
(5)remove autotools related data. (after 4.7 release?)

### PR DESCRIPTION
**Pull request by usamimikamen-sk**
depends on PR #215.
---
Remove autotools and related distribution settings.

This PR depends on PR #215.

---
 65dab67 rebase

### CODE REVIEW

**Comment by derekatkins** [comment]
If (when?) this gets merged into maint, we need to coordinate because this *WILL* break the nightly builds on `code` which currently use the auto-tools build system.

**Comment by gjanssens** [comment]
> If (when?) this gets merged into maint, we need to coordinate because this _WILL_ break the nightly builds on `code` which currently use the auto-tools build system.

You don't have to wait for this PR to get merged.  cmake support has been complete  for some time now. So  you  can switch to it beforehand. Unless our  nightlies run `make clean` (to become `ninja clean`) as that is the topic of PR #216 .

**Comment by jralls** [comment]
> You don't have to wait for this PR to get merged. cmake support has been complete for some time now. So you can switch to it beforehand. Unless our nightlies run make clean (to become ninja clean) as that is the topic of PR #216 .

@derekatkins can use cmake with make (don't say `-G Ninja` in the cmake command) and avoid the ninja clean bug.

**Comment by derekatkins** [comment]
@jralls --
> @derekatkins can use cmake with make (don't say `-G Ninja` in the cmake command) and avoid the ninja clean bug.

Currently the build server does not have cmake installed.  I can install cmake 3.14.5, provided that version works for us.

**Comment by jralls** [comment]
@derekatkins 3.14 is fine. The minimum is 3.5 for gnucash-docs and 3.10 for gnucash itself.

File: cmake/AddGHelpTarget.cmake [hunk 1]
```diff
@@ -77,6 +93,13 @@ function (add_ghelp_target docname lang entities figures dtd_files)
 
 
     install(DIRECTORY "${BUILD_DIR}"
-        DESTINATION "share/gnome/help/${docname}")
+        DESTINATION "share/help/${lang}")
+
+    # Cleaning
+    # ADDITIONAL_CLEAN_FILES is introduced in cmake 3.15 but we can't use it.
```
Comment (jralls) [inline cmake/AddGHelpTarget.cmake:99]: Please say instead "Can be replaced with ADDITIONAL_CLEAN_FILES once we can adopt Cmake 3.15."

**Comment by jralls** [comment]
That last commit does a lot more than add an extra clean target. Please break it up into separate commits for each task and include in the commit message the reason for each. As you do that consider whether the change is really part of removing autotools or really belongs in a separate PR.

What's the point of cramming HACKING and CmakeNotes.txt into README.md?

**Comment by usamimikamen-sk** [comment]
> That last commit does a lot more than add an extra clean target. Please break it up into separate commits for each task and include in the commit message the reason for each. As you do that consider whether the change is really part of removing autotools or really belongs in a separate PR.

I'm sorry. I will rearrange the branches and PRs.

> What's the point of cramming HACKING and CmakeNotes.txt into README.md?

Both HACKING and CmakeNotes.txt are outdated. I'd like to direct contributors to the Wiki.

File: cmake/AddGHelpTarget.cmake (hunk 1 above)
Comment (usamimikamen-sk) [reply cmake/AddGHelpTarget.cmake:99]: I read the ADDITIONAL_CLEAN_FILES document and found they are not enough functionality. So I will change this code.

**Comment by usamimikamen-sk** [comment]
I only picked up autotools removal and related changes.

**Comment by jralls** [comment]
> Both HACKING and CmakeNotes.txt are outdated. I'd like to direct contributors to the Wiki.

If they're obsolete they should be deleted, not made more prominent in the file that displays under the directory listing on Github.

**Comment by fellen** [comment]
I would prefer to drop autotools after the release.
Do you have parts you want to get added before?

**Comment by jralls** [comment]
I don't see any rush to do any of this before the release.

**Pull request closed without merging**